src/components/progress-report/workflow/progress-report-workflow.repository.ts

```diff
@@ -146,4 +152,79 @@ export class ProgressReportWorkflowRepository extends DtoRepository(
       changes: { status },
     });
   }
+
+  async getProjectMemberInfoByReportId(reportId: ID) {
+    const query = this.db
+      .query()
+      .match([
+        node('report', 'ProgressReport', { id: reportId }),
+        relation('in', '', ACTIVE),
+        node('engagement', 'Engagement'),
+        relation('in', '', 'engagement', ACTIVE),
+        node('project', 'Project'),
+        relation('out', '', 'member', ACTIVE),
+        node('member', 'ProjectMember'),
+        relation('out', '', 'user', ACTIVE),
+        node('user', 'User'),
+      ])
+      .optionalMatch([
+        node('user'),
+        relation('out', '', 'email', ACTIVE),
+        node('email', 'EmailAddress'),
+      ])
+      .match([
+        node('member'),
+        relation('out', '', 'roles', ACTIVE),
+        node('role', 'Property'),
+      ])
+      .return<{
+        id: ID;
+        email?: string;
+        role?: Role[];
+      }>(['user.id as id', 'email.value as email', 'role.value as role']);
+    return await query.run();
+  }
+
+  async getUserInfoByEmail(email: string) {
+    const query = this.db
+      .query()
+      .match([
+        node('email', 'EmailAddress', { value: email }),
+        relation('in', '', 'email', ACTIVE),
+        node('user', 'User'),
+      ])
+      .return<{
+        id?: ID;
+        email?: string;
```

Comment:
I don't think either of these are nullable.
```suggestion
        id: ID;
        email: string;
```
Email is being matched by so it definitely exists based on input.
`User.id` always exists, if the user/email is not found there is just no row returned.